src/compiler/crystal/macros/interpreter.cr

```diff
@@ -363,9 +363,14 @@ module Crystal
         end
 
         args = node.args.map { |arg| accept arg }
+        named_args = Hash(String, ASTNode).new
```

Comment:
could this be nilable and only created when named args is present?

Reply:
doing that also means we can pass nil in a couple of places below

Reply:
I suppose we could, but would mean that the macro methods would need each handle nil.  I thought it was a better UX if it's just empty.  Similar to how the `args` is just an empty array every time vs nil.

Reply:
Named args are very uncommon, unlike args. That's why they are nilable. If we don't make them nilable it means. hash allocation for every macro method being evaluated. I'd rather avoid allocations if possible.